FortranCon2021-stdlib/stdlib-talk.tex

```diff
@@ -0,0 +1,302 @@
+  \and Evan Voyles}
+\date{FortranCon 2021 - 24 Sept 2021}
+
+\begin{document}
+
+\begin{frame}
+  \titlepage
+\end{frame}
+
+\begin{frame}
+  \frametitle{What is \sout{new in} stdlib?}
+
+  \begin{itemize}
+  \item Part of fortran-lang: https://github.com/fortran-lang/stdlib
+  \item Bridge the gap between intrinsics and applications
+    \begin{description}[Mathematics]
+    \item[Utilities] strings, logging, filesystem interaction
+    \item[Algorithms] searching, sorting
+    \item[Mathematics] linear algebra, special functions, statistics
+    \end{description}
+  \item Somewhere between C++ stdlib \& Matlab/scipy
+  \end{itemize}
+
+  \note{
+    Before I can show off what's new in the standard library, let's have a few words about what it is.
+    It's a volunteer effort maintained under the fortran-lang organization, with the code hosted on Github.
+    The main goal is to bridge the gap between what's offered intrinsically in the language versus the high-level code one would like to write in applications.
+    This makes the scope of the Fortran standard library quite wide.
+    We want to offer general programming-type things like common algorithms a la C++, while also providing a good base of numerically-oriented functionality like what's found in Matlab or the Python scientific ecosystem.
+    That's a tall order, but we've made good headway.
+  }
+\end{frame}
+
+
+\begin{frame}
+  \frametitle{stdlib has roughly doubled in size in the past year}
+
+  \begin{block}{Modules one year ago vs today}
+    \centering
+    \begin{tabular}{ccc}
+      ascii &
+      \textcolor{red}{bitsets} &
+      error \\
+      io &
+      kinds &
+      linalg \\
+      \textcolor{red}{logger} &
+      \textcolor{red}{math} &
+      optval \\
+      quadrature &
+      \textcolor{red}{sorting} &
+      \textcolor{red}{specialfunctions} \\            
+      stats &
+      \textcolor{red}{stats\_distribution\_PRNG} &
+      \textcolor{red}{stringlist\_type} \\
+      \textcolor{red}{strings} &
+      \textcolor{red}{string\_type} &                           
+      system 
+    \end{tabular}
+  \end{block}
+
+  \note{
+    By raw count of modules included in stdlib, the project has roughly doubled in size in the past year.
+    A year ago, stdlib looked more like a grab-bag of utilities than a standard library.
+    Now, it looks much more deserving of its name, with lots of exciting development in general-purpose programming functionality like string handling, sorting, logging, and support for bitsets.
+    Let's have a look at some examples.
+  }
+\end{frame}
+
+
+\begin{frame}[fragile]
+  \frametitle{Demo: stdlib\_logger}
+
+
+  \begin{block}{ex\_logger.f90}
+    \small
+\begin{verbatim}
+use stdlib_logger, only: global_logger
+implicit none
+call global_logger%add_log_file('log.txt')
+call global_logger%log_debug('I am invisible')
+call global_logger%log_information('Something informative')
+call global_logger%log_error('Oopsie daisy')
+end
```

Comment:
Maybe make use of `lstlisting` to get some basic syntax hints in (mainly by typeface not color). For the fpm presentation I highlighted the code snippets externally and included them as pdf.

Reply:
Hm, I tried mimicking the example in the beamer manual and made a mess of it. In your opinion are the code listings hard to read without any styling?